lib/model/folder_sendrecv.go

```diff
@@ -508,15 +508,20 @@ nextFile:
 		devices := snap.Availability(fileName)
 		for _, dev := range devices {
 			if f.model.ConnectedTo(dev) {
-				// Handle the file normally, by copying and pulling, etc.
-				f.handleFile(fi, snap, copyChan)
-				continue nextFile
+				// Check if the remote folder state is valid on connected device (i.e. not paused)
+				if state, ok := f.model.remoteFolderStates[dev][f.folderID]; ok {
```

Comment:
You cannot access `remoteFolderStates` without holding a read lock on the mutex in model. And it feels a bit messy to do that from here. I'd create a utility function in model for this. Respectively it looks like it almost already exists `model.availabilityInSnapshotRLocked` - just that it's for a block instead of the entire file. You could split that into two functions, one for the part with the snapshot, one with the devicedownloads. Then use the snapshot function here. Ok that explanations sounds unintelligible on re-reading, maybe some pseudo-code is clearer:

Split functions in model, and new "wrapper" function (also need to adapt callers to the split functions, leaving that out):

```
func (m *model) availabilityInSnapshot(cfg config.FolderConfiguration, snap *db.Snapshot, file protocol.FileInfo) []Availability {
	m.mut.RLock()
	defer m.mut.RUnlock()
	return m.availabilityInSnapshotRLocked(...)
}

func (m *model) availabilityInSnapshotRLocked(cfg config.FolderConfiguration, snap *db.Snapshot, file protocol.FileInfo) []Availability {
	var availabilities []Availability
	for _, device := range snap.Availability(file.Name) {
...
	return availabilities
}

func (m *model) availabilityFromTemporaryRLocked(cfg config.FolderConfiguration, file protocol.FileInfo, block protocol.BlockInfo) []Availability {
	var availabilities []Availability
	for _, device := range cfg.Devices {
...
	return availabilities
}
```

And then here just:

```
if devices := f.model.availabilityInSnapshot(...); len(devices) > 0 {
f.handleFile(...)
}
```

Does that make any sense?

Reply:
Yup, Sounds good!
will look at this tomorrow.

Reply:
Hi again @hireworksltd. This PR just came up because of a potentially related support request, thus reminded me of it - forgot about it before. No pressure or anything, just a reminder in case it's been forgotten in the hopes to be helpful (at least to me it usually is :) ).

Reply:
Hi Simon,

Sorry been a bit mad here last few weeks! Will try set aside some time over the next few days.

Best,
Jon

Reply:
Thanks for the update - all good, no pressure. Also the original offer still stands, that I could do/finish the change if you'd rather spend your time on something else. (I hope that doesn't come across as passive-aggressiv or push, really just an offer :) )

Reply:
Yes that looks good, thanks. There's a few things that look off, but they are small, the general approach holds up. If you push those changes to the branch here (or create a new PR), I can point those things out with comments inline - simpler than trying to describe them.

Reply:
Many thanks!